Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -2407,6 +2408,9 @@ void LocalFrameView::maintainScrollPositionAtAnchor(ContainerNode* anchorNode)
 
     cancelScheduledScrolls();
 
+    if (is<Element>(anchorNode))
+        m_frame->document()->contentVisibilityDocumentState().updateContentRelevancyStatusForScrollIfNeeded(downcast<Element>(*anchorNode));
```

Comment:
Forcing creation of ContentVisibilityDocumentState here again.

Reply:
Fixed.